tests/playwright-test/global-setup.spec.ts

```diff
@@ -386,3 +386,67 @@ test('teardown after error', async ({ runInlineTest }) => {
     'teardown 1',
   ]);
 });
+
+function expectInOrder(output: string, tokens: string[]) {
+  for (const token of tokens)
+    expect(output).toContain(token);
+
+  const positions = tokens.map(token => `${output.indexOf(token).toString().padStart(3, '0')} ${token}`);
+  expect(positions, 'order is correct').toEqual(positions.toSorted());
+}
+
+test('globalSetup should support multiple', async ({ runInlineTest }) => {
+  const { passed, output } = await runInlineTest({
+    'playwright.config.ts': `
+      module.exports = {
+        globalSetup: ['./globalSetup1.ts','./globalSetup2.ts','./globalSetup3.ts'],
+        globalTeardown: ['./globalTeardown1.ts', './globalTeardown2.ts'],
+      };
+    `,
+    'globalSetup1.ts': `module.exports = () => console.log('globalSetup1');`,
+    'globalSetup2.ts': `module.exports = () => console.log('globalSetup2');`,
+    'globalSetup3.ts': `module.exports = () => console.log('globalSetup3');`,
+    'globalTeardown1.ts': `module.exports = () => console.log('globalTeardown1');`,
+    'globalTeardown2.ts': `module.exports = () => console.log('globalTeardown2');`,
+
+    'a.test.js': `
+      import { test } from '@playwright/test';
+      test('a', () => console.log('test a'));
+      test('b', () => console.log('test b'));
+    `,
+  }, { reporter: 'line' });
+
+  expect(passed).toBe(2);
+  expectInOrder(output, [
+    'globalSetup1',
+    'globalSetup2',
+    'globalSetup3',
+    'test a',
+    'test b',
+    'globalTeardown2',
+    'globalTeardown1',
+  ]);
+});
+
+test('debug logs show index of setup/teardown', async ({ runInlineTest }) => {
```

Comment:
I'd remove this test, it does not test an essential part of our test runner.

Instead, it would be great to test what exactly do we run in what order, when there are multiple global setups and teardowns, including teardown functions returned from global setups, and some of them fail.

Reply:
I've removed the test in https://github.com/microsoft/playwright/pull/32955/commits/4316a0178dcf0f70afc700ec1fe8a4ea620f2e73 and and added lots of coverage in https://github.com/microsoft/playwright/pull/32955/commits/3ef8178cca9b70d4b9df39f4374e663750d8ae91. Also amended the PR description with my thoughts around ordering.